spec/services/users/login_service_spec.rb

```diff
@@ -3,33 +3,66 @@
 require 'rails_helper'
 
 RSpec.describe Users::LoginService do
-  subject(:service_response) { described_class.new(email: email, password: password).execute }
+  subject(:service_response) { described_class.new(**params).execute }
 
   context 'when the credentials are correct' do
     let(:email) { generate(:email) }
+    let(:username) { generate(:username) }
     let(:password) { generate(:password) }
-    let!(:current_user) { create(:user, email: email, password: password) }
+    let!(:current_user) { create(:user, email: email, username: username, password: password) }
 
-    it 'returns a success response' do
-      expect(service_response).to be_success
-      expect(service_response.payload).to be_a(UserSession)
-      expect(service_response.payload.user.id).to eq(current_user.id)
-      expect(service_response.payload.token).to be_present
+    def check_audit_events(user, key, other_key)
       is_expected.to create_audit_event(
         :user_logged_in,
-        author_id: current_user.id,
+        author_id: user.id,
         entity_type: 'User',
-        entity_id: current_user.id,
-        details: { email: current_user.email, method: 'username_and_password' },
+        entity_id: user.id,
+        details: { key => current_user.send(key), method: 'username_and_password' },
         target_type: 'User',
-        target_id: current_user.id
+        target_id: user.id
       )
+      is_expected.not_to create_audit_event(
+        :user_logged_in,
+        details: { other_key: current_user.send(other_key), method: 'username_and_password' }
```

Comment:
This does make a difference. With `other_key:` you expect the audit event to have a `other_key` key in the details, while `other_key =>` expects a key with the value of `other_key` in the details.
```suggestion
        details: { other_key => current_user.send(other_key), method: 'username_and_password' }
```